03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
+>the location in our names (eg. `DSM` not `DSM_HARV`).  This hasn't gotten us 
+>in trouble as we've only been working in one location (Harvard Forest), 
+>however, it is still good practice to be specific with file and object names.
+
+When we write this raster object to a GeoTIFF file we'll name it 
+`chm_ov_HARV.tiff` this way we know that it is a canopy height model of Harvard 
+Forest created using overlay.  Any files you write will be saved in whichever 
+file is the current working directory. 
 
 ```{r write-raster }
-
-#export CHM object to new geotiff
-writeRaster(CHM_ov_HARV,"chm_ov_HARV.tiff",
+#export CHM object to new GeotIFF
+writeRaster(CHM_ov,"chm_ov_HARV.tiff",
             format="GTiff", 
             overwrite=TRUE, 
             NAflag=-9999)
-
-#note we are setting the NA value to -9999 for consistency.
 ```
 
-#Challenge 1
-1. Import the DSM and DTM from the SJER directory. These data are also derived from
-NEON lidar data but cover the San Joachin experimental range field site in 
-Southern California. Create a CHM from the two raster layers. Export as a geotiff.
-{: .notice }
+In the code we specify that the format will be GTiff. Overwrite is true so that
+it replaces any other file with this name.  This is a useful setting so that you
+can make a change to the object and then update the file by re-runing the code.
+However, be careful to never accidentally re-use a name for a file that already 
+exists. We are setting the NA value to -9999 for consistency. 
+####CONSISTENCY WITH WHAT?  NOT PREVIOUSLY MENTIONED
 
-```{r SJER-CHM, echo=FALSE }
 
-##load the DTM
-DTM_SJER <- raster("NEON_RemoteSensing/SJER/DTM/SJER_dtmCrop.tif")
-#load the DSM
-DSM_SJER <- raster("NEON_RemoteSensing/SJER/DSM/SJER_dsmCrop.tif")
+#Challenge: Exploring the San Joaquin field site.  
+As ecologists we may want to explore the phenology of two different sites.  The 
+San Joaquin valley in California is rather different from the Harvard Forest in 
+Massachusetts and could make a good comparison.  
 
-#use overlay to subtract the two rasters
-CHM_ov_SJER <- overlay(DSM_SJER,DTM_SJER,fun=subtRasters)
+LiDAR and imagery data used to create the rasters were collected at the San 
+Joaquin NEON field sites the same as for the Harvard Forest NEON field sites. 
 
-#plot the output
-plot(CHM_ov_SJER,
-     main="NEON Canopy Height Model - Overlay Subtract\n SJER")
+Your challenge is to import the San Joaquin raster files so that we can compare 
+to two sites. So that we can use object/file names in future lessons names 
+should follow the format of objectType_SJER (e.g. `DSM_SJER`).
 
-```
+1. Import the DSM and DTM from the SJER directory. Don't forget to examine the
+data for CRS, bad values, etc. 
+2. Create a CHM from the two raster layers and check to make sure the data are 
+what you expect. 
+3. Plot the CHM from SJER. 
+4. Export as a GeoTIFF.
+5. What do you notice about tree heights at the two sites?
 
+```{r SJER-CHM, }
```

Comment:
echo=FALSE 

I think i had the plot output with echo=FALSE so they could see what it looks like in the lesson but not the code.